contracts/contracts/mocks/curve/MockCurvePool.sol

```diff
@@ -62,47 +63,65 @@ contract MockCurvePool {
         // solhint-disable-next-line no-unused-vars
         uint256 _minAmount
     ) external {
-        IERC20(lpToken).transferFrom(msg.sender, address(this), _amount);
+        // Burn the Curve LP tokens
+        IBurnableERC20(lpToken).burnFrom(msg.sender, _amount);
         uint256[] memory amounts = new uint256[](coins.length);
         amounts[uint128(_index)] = _amount;
-        uint256 amount = calc_withdraw_one_coin(_amount, _index);
-        IERC20(coins[uint128(_index)]).transfer(msg.sender, amount);
-        // solhint-disable-next-line reentrancy
-        balances[uint128(_index)] = balances[uint128(_index)] - amount;
+        uint256 coinAmount = calc_withdraw_one_coin(_amount, _index);
+        balances[uint128(_index)] -= coinAmount;
+        IERC20(coins[uint128(_index)]).transfer(msg.sender, coinAmount);
     }
 
     function get_virtual_price() external pure returns (uint256) {
         return 1 * 10**18;
```

Comment:
yes, it would make sense but it breaks the old OUSD AMO unit tests. Given there's a big refactor of the AMOs on another branch, I think we should do this change with the AMOs.

Reply:
agreed 👍